### ISSUE TITLE
nom claims build is still running when used with `nix develop`

### ISSUE THREAD

**Issue by kubukoz** [maralorn/nix-output-monitor#81]
For example, `nix develop nixpkgs#cowsay --print-build-logs |& nom` runs forever. If you happen to have something require a rebuild, nom will show the last packages as still building, even when they're done:

<img width="267" alt="image" src="https://user-images.githubusercontent.com/894884/212095690-4a355d7d-5d4f-49b0-b6b7-6a3d326156d1.png">

**Comment by maralorn** [comment]
I don˚t think it makes sense to use nom in this way. nix develop will enter into a shell after finishing the build. But nom has no way to know which part of the output is build output and which is the shell prompt. It has no magic to detect that nix is finished and exit. While I agree that it might be a bug, that the last build is not shown as completed that might have to do with how the process exits, which has some subtleties. Please try
`nom develop nixpkgs#cowsay` and tell me if you are experiencing the same issue.

**Comment by kubukoz** [comment]
> I don't think it makes sense to use nom in this way

I agree! However, people who don't know that might be very confused when they try to, and nom pretends like it's still building.

I'm happy to close the issue if it can't be fixed here, it would still be searchable so it might help someone when they bump into this problem.

`nom develop` seems to behave correctly.

**Comment by maralorn** [comment]
I see the problem. If I think about it, I actually know what the bug is, so let’s keep this open until I haved fixed it.

It’s a problem where nom will not check if a build is finished if it doesn’t receive any input. Normally that doesn‘t matter much, because there is usually always some input, but in this case once the bash prompt was printed nothing else happens and nom actually goes stale.
